tests/modules/sightings/test_models.py

```diff
@@ -1,48 +1,101 @@
-        response = flask_app_client.post(
-            '/api/v1/sightings/',
-            data=json.dumps(
-                {
-                    'title': 'Testing generated.',
-                }
-            ),
-        )
-
-    assert response.status_code == 200
-    assert response.content_type == 'application/json'
-    assert isinstance(response.json, dict)
-    assert set(response.json.keys()) >= {
-        'guid',
-        'title',
-    }
-
-    sighting_guid = response.json['guid']
-    read_sighting = Sighting.query.get(response.json['guid'])
-    assert read_sighting.title == 'Testing generated.'
-
-    with flask_app_client.login(regular_user, auth_scopes=('sightings:read',)):
-        response = flask_app_client.get('/api/v1/sightings/%s' % sighting_guid)
-
-    assert response.status_code == 200
-    assert response.content_type == 'application/json'
-    assert isinstance(response.json, dict)
-    assert set(response.json.keys()) >= {
-        'guid',
-        'title',
-    }
-
-    with flask_app_client.login(regular_user, auth_scopes=('sightings:write',)):
-        response = flask_app_client.delete('/api/v1/sightings/%s' % sighting_guid)
-
-    assert response.status_code == 204
-    read_sighting = Sighting.query.get(sighting_guid)
-    assert read_sighting is None
+    test_sighting = Sighting(
+        title='New Test Sighting',
+    )
+
+    test_encounter_a = Encounter(
+        title='New Test Encounter A',
+    )
+
+    test_encounter_b = Encounter(
+        title='New Test Encounter B',
+    )
+
+    with db.session.begin():
+        db.session.add(test_encounter_a)
+        db.session.add(test_sighting)
+
+    db.session.refresh(test_encounter_a)
+    db.session.refresh(test_sighting)
+
+    assert len(test_sighting.get_encounters()) == 0
+    assert test_encounter_a.get_sighting() is None
+
+    test_sighting.add_encounter(test_encounter_a)
+
+    with db.session.begin():
+        db.session.add(test_encounter_a)
+        db.session.add(test_sighting)
+
+    assert len(test_sighting.get_encounters()) == 1
+    assert test_encounter_a.get_sighting() is not None
+
+    test_sighting.add_encounter(test_encounter_b)
+
+    with db.session.begin():
+        db.session.add(test_encounter_b)
+        db.session.add(test_sighting)
+
+    # making sure i didn't accidentally set up a 1 to 1
+    assert len(test_sighting.get_encounters()) == 2
+
+    logging.info(test_sighting.get_encounters())
+    logging.info(test_encounter_a.get_sighting())
+
+
+
+def test_sighting_ensure_no_duplicate_encounters(db):
+    from app.modules.sightings.models import Sighting
+    from app.modules.encounters.models import Encounter
+
+    test_sighting = Sighting(
+        title='New Test Sighting',
+    )
+
+    test_encounter = Encounter(
+        title='New Test Encounter',
+    )
+
+    with db.session.begin():
+        db.session.add(test_encounter)
+        db.session.add(test_sighting)
+
+    db.session.refresh(test_encounter)
```

Comment:
I might be wrong but I think you need the with db.session.begin() bit for refreshes as well.